pkg/openthread/contrib/sock/udp/openthread_sock_udp.c

```diff
@@ -0,0 +1,387 @@
+	sock->data = NULL;
+	sock->len = payload_len;
+
+	// msg_t msg = { .type = _MSG_TYPE_RCV };
+	msg_t msg = { .type = 1 };
+
+    //mutex_lock(&sock->mutex);
+    /*sock->recv_info.src_port = src_port;
+    sock->recv_info.src = (const ipv6_addr_t *)src_addr;
+    sock->recv_info.data = data;
+    sock->recv_info.datalen = datalen - sizeof(ipv6_hdr_t);*/
+    //mutex_unlock(&sock->mutex);
+    mbox_put(&sock->mbox, &msg);
+}
+
+/**
+ * @brief   Internal helper functions for OPENTHREAD
+ * @internal
+ * @{
+ */
+/**
+ * @brief   Checks if address family is not supported
+ * @internal
+ */
+static inline bool openthread_af_not_supported(int af)
+{
+    /* TODO: add AF_INET support */
+    return (af != AF_INET6);
+}
+
+/**
+ * @brief   Check if end point points to any address
+ * @internal
+ */
+static inline bool openthread_ep_addr_any(const sock_ip_ep_t *ep)
+{
+    assert(ep != NULL);
+    const uint8_t *p = (uint8_t *)&ep->addr;
+    for (uint8_t i = 0; i < sizeof(ep->addr); i++) {
+        if (p[i] != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+/*
+static OT_JOB _set_panid(otInstance *ot_instance, void *data)
+{
+    uint16_t panid = *((uint16_t*) data);
+    otLinkSetPanId(ot_instance, panid);
+}
+
+static OT_JOB _get_panid(otInstance *ot_instance, void *data)
+{
+    *((uint16_t*) data) = otLinkGetPanId(ot_instance);
+    printf("PanID: %04x\n", *((uint16_t*) data));
+}
+
+static OT_JOB _thread_start(otInstance *ot_instance, void *data)
+{
+    printf("Starting OpenThread\n");
+    otIp6SetEnabled(ot_instance, true);
+    otThreadSetEnabled(ot_instance, true);
+}
+
+static OT_JOB _read_state(otInstance *ot_instance, void *data)
+{
+    uint8_t state = otThreadGetDeviceRole(ot_instance);
+    printf("State is: %i\n", state);
+}
+
+static OT_JOB _get_ip_addresses(otInstance *ot_instance, void *data)
+{
+    for(const otNetifAddress *addr=otIp6GetUnicastAddresses(ot_instance); addr; addr=addr->mNext)
+    {
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+        printf("inet6 %s\n", ipv6_addr_to_str(addrstr, (ipv6_addr_t*) &addr->mAddress.mFields, sizeof(addrstr)));
+    }
+}
+*/
+static OT_JOB _create_udp_socket(otInstance *ot_instance, void *data)
+{
+    sock_udp_t *sock = (sock_udp_t*) data;
+    otSockAddr sockaddr;
+    memset(&sockaddr, 0, sizeof(otSockAddr));
+    sockaddr.mPort = *((uint16_t*) &(sock->local.port));
+
+    otUdpOpen(ot_instance, &sock->ot_udp_socket, _handle_receive, sock);
+    otUdpBind(&sock->ot_udp_socket, &sockaddr);
+}
+
+static OT_JOB _send_udp_pkt(otInstance *ot_instance, void *data)
+{
+    sock_udp_t *sock = (sock_udp_t*) data;
+    udp_pkt_t *pkt = (udp_pkt_t*) &(sock->pkt);
+    otMessage *message;
+
+    message = otUdpNewMessage(ot_instance, true);
+    int error;
+    error = otMessageSetLength(message, pkt->len);
+    error = otMessageWrite(message, 0, pkt->payload, pkt->len);
+    (void) error;
+
+    otMessageInfo mPeer;
+    
+    //Set dest address
```

Comment:
changed to C style comments everywhere except in the unfinished part. Those won't be there when the sock implementation is 100% finished